### PR TITLE
fuzz: fixes oss-fuzz: 9895

### PR DESCRIPTION
**Pull request by anirudhmurali**
Title: Fixes oss-fuzz: 9895

Description: oss-fuzz issue (9895): https://oss-fuzz.com/v2/testcase-detail/6195059702628352
The given input testcase had multiple `START_TIME` formatters as value, a valid input must contain only one `START_TIME`, throwing exception if invalid.

Risk Level: Low

Testing: Tested unit tests (bazel test //test/common/router:header_parser_fuzz_test), built and ran fuzzers with oss-fuzz.

Signed-off-by: Anirudh M m.anirudh18@gmail.com

### CODE REVIEW

File: source/common/router/header_formatter.cc [hunk 1]
```diff
@@ -143,7 +143,10 @@ RequestInfoHeaderFormatter::RequestInfoHeaderFormatter(absl::string_view field_n
     }
     field_extractor_ = [this, pattern](const Envoy::RequestInfo::RequestInfo& request_info) {
       const auto& formatters = start_time_formatters_.at(pattern);
-      ASSERT(formatters.size() == 1);
+      if (formatters.size() != 1) {
```
Comment (htuch) [inline source/common/router/header_formatter.cc]: @dio does this seem right?

File: test/common/router/header_parser_corpus/clusterfuzz-testcase-header_parser_fuzz_test-6195059702628352 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+headers_to_add {
```
Comment (htuch) [inline test/common/router/header_parser_corpus/clusterfuzz-testcase-header_parser_fuzz_test-6195059702628352]: @anirudhmurali can you add a test case to the unit tests for `header_formatter.cc`? Best practice is to check in both corpus and a unit test for the new error handling.

File: source/common/router/header_formatter.cc [hunk 3]
```diff
@@ -143,7 +143,10 @@ RequestInfoHeaderFormatter::RequestInfoHeaderFormatter(absl::string_view field_n
...
+        throw EnvoyException(
+            fmt::format("multiple start_time values found: {}", formatters.size()));
```
Comment (dio) [inline source/common/router/header_formatter.cc]: @anirudhmurali, thanks for checking on this.

Questions: 

1. Do you think "multiple %START_TIME% patterns..." sounds better? 
2. Is the input here: https://github.com/envoyproxy/envoy/pull/4189/files#diff-084669b1d67ce3df5fbda83bef9354bbR3 causing this?

File: source/common/router/header_formatter.cc (hunk 1 above)
Comment (dio) [reply source/common/router/header_formatter.cc]: On the second thought, I think we can support multiple patterns. 

@anirudhmurali Do you want to take it?

File: source/common/router/header_formatter.cc (hunk 3 above)
Comment (dio) [reply source/common/router/header_formatter.cc]: NVM, I tried to patch the `header_formatter` to accept this pattern from the corpus, and it seems it makes sense to have multiple patterns inline as the header value. 

@anirudhmurali, as I said above I can work with you to fix this.

(This fuzzer is really good then. This is my mistake, sorry).

**Comment by anirudhmurali** [comment]
@dio With regards to the Slack discussion and the GitHub comments, I believe this allows definition of multiple `START_TIME` patterns. Have added the case to the test file as well. Please let me know if there is anything more to this. Thanks.

File: test/common/router/header_formatter_test.cc [hunk 4]
```diff
@@ -617,6 +617,10 @@ match: { prefix: "/new_endpoint" }
         key: "x-request-start"
         value: "%START_TIME(%s.%3f)%"
       append: true
+    - header:
+        key: "x-request-start"
```
Comment (dio) [inline test/common/router/header_formatter_test.cc]: Let's rename the key to something else, e.g. `"x-request-start-multiple"`, and add some similar expectations below.

```
EXPECT_TRUE(headerMap.has("x-request-start-multiple"));
EXPECT_EQ("1522796769.123 2018-04-03T23:06:09.123Z 1522796769");
```

**Pull request merged**